src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -609,9 +609,53 @@ public unsafe void Reset()
         {
             _updateStopwatch.Stop();
             _renderStopwatch.Stop();
-            _glfw.DestroyWindow(_windowPtr);
+
+            try
+            {
+                _glfw.DestroyWindow(_windowPtr);
+            }
+#pragma warning disable 168
+            catch(GlfwException e)
+#pragma warning restore 168
+            {
+                // If the window is already destroyed, it throws an exception,
+                // but we want the window destroyed anyways, so just ignore it
+            }
+            
             _windowPtr = (WindowHandle*) 0;
         }
+        
+        // Disable parameter because 
+        // ReSharper disable once UnusedParameter.Local
+        private void Dispose(bool disposing)
+        {
+            Reset();
+            
+            // All callbacks are initialized at the same time,
+            // so checking each one individually shouldn't be
+            // necessary.
+            if (_onClosing != null)

```

Comment:
Hmmm in future you should probably prefer `is null` rather than `== null` as it can do weird things if the operator is improperly implemented. Luckily though, it doesn’t matter in this case it’s just something to keep in mind.